src/raftstore/store/snap.rs

```diff
@@ -61,10 +115,82 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub trait SendSnapshotFileReader: Read {
+    fn exists(&self) -> bool;
+    fn path(&self) -> String;
+    fn meta(&self) -> io::Result<Metadata>;
+    fn total_size(&self) -> io::Result<u64>;
+    fn delete(&self);
+}
+
+pub trait RecvSnapshotFileReader {
+    fn path(&self) -> String;
```

Comment:
`Path::display` is evaluated lazily. Maybe we need a method named `display_path`.